mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -107,9 +107,15 @@ class BisectingKMeansModel private[ml] (
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     transformSchema(dataset.schema, logging = true)
-    val predictUDF = udf((vector: Vector) => predict(vector))
-    dataset.withColumn($(predictionCol),
-      predictUDF(DatasetUtils.columnToVector(dataset, getFeaturesCol)))
+    if ($(predictionCol).nonEmpty) {
+      val predictUDF = udf((vector: Vector) => predict(vector))
+      dataset.withColumn($(predictionCol),
+        predictUDF(DatasetUtils.columnToVector(dataset, getFeaturesCol)))
+    } else {
+      this.logWarning(s"$uid: BisectingKMeansModel.transform() was called as NOOP" +
```

Comment:
PS we should fix these messages. "called as NOOP" is cryptic. Just say "transform() does nothing because ..."